Otus-DevOps-2018-02/ifqthenp_microservices
# HW 23. Application Deploy to the GKE Cluster and k8s Security Model (kubernetes-2)

# Completed Homework №23


## The following tasks have been completed

- :large_blue_diamond: Created k8s Deployment, Service, and Namespace kinds config files for Reddit application deployment
- :large_blue_diamond: Application has been successfully deployed to the local `minikube` cluster
- :large_blue_diamond: GKE cluster with 2 nodes has been created using GCP web user interface
- :large_orange_diamond: GKE cluster with 2 nodes has been created using `terraform` tool
- :large_orange_diamond: Created config file allowing `kubernetes-dashboard` service account to access information about the cluster

```yml
---
apiVersion: rbac.authorization.k8s.io/v1beta1
kind: ClusterRoleBinding
metadata:
  name: kubernetes-dashboard
  labels:
    k8s-app: kubernetes-dashboard
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: cluster-admin
subjects:
- kind: ServiceAccount
  name: kubernetes-dashboard
  namespace: kube-system
```

## How to get the project running

1. Install `kubectl` and `minikube` command line tools
2. Run `minikube  start`
3. `cd` into `kubernetes/reddit` directory
4. Run `kubectl apply -f dev-namespace.yml`
5. Run `kubectl apply -n dev -f .`
6. Run `minikube service ui -n dev`

## How to check that the project actually works

- Ensure that the build passed TravisCI tests
  - [#andrei_bogomja](https://devops-team-otus.slack.com/messages/C9M6L1XB3) Slack channel
  - actual GitHub pull request page
- Having successfully deployed Reddit app to the GKE cluster, run:
  - `kubectl get nodes -o wide`
  - `kubectl describe service ui -n dev | grep NodePort
  - Type in browser' URL bar `http://<node-ip>:<NodePort>`

![alt text](https://raw.githubusercontent.com/ifqthenp/files/master/gke_screenshot.png "Reddit app on GKE cluster")

## PR checklist
